### PR TITLE
feat: move logic from generate.js to api-endpoint

### LINKED ISSUE #923: [Feature] Move `generate.js` to API-Endpoint
We should definitely add the logic form `generate.js` to the API-Endpoint form which the `index` gets its data from. By doing so we can possibly connect other UIs to this API-Endpoint

### PR DESCRIPTION
**Pull request by Cahllagerfeld**
closes #923 

## Definition of Done

- [ ] remove `generate.js`
- [ ] remove `generate script`
- [ ] move logic from existing file to api-endpoint

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/awesome-github-profiles/pull/928"></a>

### CODE REVIEW

File: pages/api/profiles.js [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+import fs from "fs";
+import path from "path";
 
-export default function handler(req, res) {
-  res.status(200).json(data)
+export default async function handler(req, res) {
+  const relativeDir = path.resolve("./public", "profiles");
+  const files = fs.readdirSync(relativeDir);
+
+  const profiles = await Promise.all(
```
Comment (Cahllagerfeld) [inline pages/api/profiles.js:8]: This is a bit fragile imo. If one Promise fails, `Promise.resolve` wont resolve successfully. I think a propper error-handling needs to be added here. Its discussable if with this, or a future PR.

File: pages/api/profiles.js [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+    files.map(async (file) => {
+      const response = await fetch(
+        `http://localhost:3000${path.join("/", "profiles", file)}`
```
Comment (Cahllagerfeld) [inline pages/api/profiles.js:11]: We should maybe think of a solution of not hardcoding the URL although it needs to be a absolute one.
(Afaik, in index.js the Request-URL is also hardcoded.)

**Review by Cahllagerfeld — COMMENTED** [review]
pointed out to spots which are a bit fragile, but worked in my tests

File: pages/api/profiles.js (hunk 2 above)
Comment (eddiejaoude) [reply pages/api/profiles.js:11]: I think if a request url the `join` is not required, only if it is a disk path

I wonder if we can use a relative url here? `/profiles/...`

**Comment by eddiejaoude** [comment]
I am going to merge this and more work can continue in the `nextjs-live` branch (or via another PR if preferred)

**Pull request merged**